python/tk_multi_reviewsubmission/renderer.py

```diff
@@ -1,187 +1,210 @@
+        }
+
+        render_info = {
+            'burnin_nk': self._burnin_nk,
+            'slate_font': self._font,
+            'codec_settings': {'quicktime': writenode_quicktime_settings},
+        }
+
+        # set needed paths and force them to use forward slashes for use in Nuke (latter needed for Windows)
+        src_frames_path = path.replace('\\', '/')
+        movie_output_path = output_path.replace('\\', '/')
+
+        nuke_render_info = {
+            'width': width,
+            'height': height,
+            'first_frame': first_frame,
+            'last_frame': last_frame,
+            'version': version,
+            'name': name,
+            'color_space': color_space,
+            'nuke_exe_path': nuke_exe_path,
+            'render_script_path': render_script_path,
+            'shotgun_context': shotgun_context,
+            'app_settings': app_settings,
+            'render_info': render_info,
+            'src_frames_path': src_frames_path,
+            'movie_output_path': movie_output_path,
+        }
+
+        return nuke_render_info
+
+    def render_movie_in_nuke(self, path, output_path,
+                             width, height,
+                             first_frame, last_frame,
+                             version, name,
+                             color_space,
+                             active_progress_info=None):
+
+        render_info = self.gather_nuke_render_info(path, output_path, width, height, first_frame,
+                                                   last_frame, version, name, color_space)
+        # TODO: can we offload to a thread, similar to submitter?
+        run_in_batch_mode = True if nuke is None else False
+        if run_in_batch_mode:
+            # --------------------------------------------------------------------------------------------
+            #
+            #  Running within Nuke interactive ...
+            #
+            # --------------------------------------------------------------------------------------------
+
+            # Set-up the subprocess command and arguments
+            cmd_and_args = [
+                render_info.get('nuke_exe_path'), '-t', render_info.get('render_script_path'),
+                '--path', render_info.get('src_frames_path'),
+                '--output_path', render_info.get('movie_output_path'),
+                '--width', str(width), '--height', str(height), '--version', str(version), '--name',
+                name,
+                '--color_space', color_space,
+                '--first_frame', str(first_frame),
+                '--last_frame', str(last_frame),
+                '--app_settings', str(render_info.get('app_settings')),
+                '--shotgun_context', str(render_info.get('shotgun_context')),
+                '--render_info', str(render_info.get('render_info')),
+            ]
+
+            p = subprocess.Popen(cmd_and_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Please confirm that this picks up environment variables from the parent process. For example, all the DD_* variables, such as DD_SHOW/DD_SEQ/DD_SHOT and DD_WITH_OVERRIDE and DD_TEST_BRANCHES are required to run properly.

Reply:
Looks like the default behaviour of subprocess is to pass on the current process' environment. Verified this as well.
I only mentioned this because tk-multi-rendermovie was explicitly passing of env as an argument, doing os.environ.copy() and adding a few things. This was causing some symbol not found errors related to Qt .so files. Not sure why that was happening, or why it went away with removing the env argument. Will check that again later.